src/components/Editor/index.tsx

```diff
@@ -211,6 +220,105 @@ const Editor = () => {
         });
     }, []);
 
+    const setEditorSelection = useCallback((
+        selectedLogEventNum:number,
+        beginLineNumToLogEventNumMapRef:React.RefObject<BeginLineNumToLogEventNumMap>
+    ) => {
+        const selectedLogEventLineNum =
+            getMapKeyByValue(beginLineNumToLogEventNumMapRef.current, selectedLogEventNum);
+        const nextLogEventLineNum =
+            getMapKeyByValue(beginLineNumToLogEventNumMapRef.current, selectedLogEventNum + 1);
+
+        if (null === selectedLogEventLineNum) {
+            // logEventNum is not found on the current page.
+            return;
+        }
+
+        let endLineNumber: number;
+        if (null !== nextLogEventLineNum) {
+            endLineNumber = nextLogEventLineNum - 1;
+        } else {
+            // Handle the case when this is the last log event in the file.
+            const model = editorRef.current.getModel();
+            if (null === model) {
+                throw new Error("Unable to get the text model.");
+            }
+            endLineNumber = model.getLineCount() - 1;
+        }
+
+        const selectionRange = new monaco.Range(
+            selectedLogEventLineNum,
+            0,
+            endLineNumber,
+            Infinity
+        );
+
+        editorRef.current.setSelection(selectionRange);
```

Comment:
for this to work, it seems we need to comment out this line in `goToPositionAndCenter()`:
```
const goToPositionAndCenter = (
    editor: monaco.editor.IStandaloneCodeEditor,
    position: monaco.IPosition,
) => {
    editor.revealLineInCenter(position.lineNumber);

    // editor.setPosition(position);
    editor.focus();
};
```